envoy: Use TypedConfig for Envoy filters

jrajahalme opened the pull request:
Generic protobuf structpb config encoding is deprecated by
Envoy. Change to using TypedConfig to avoid flooding the logs with
deprecation warnings.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9889)
<!-- Reviewable:end -->

raybejjani commented on pkg/envoy/server.go:
```diff
@@ -174,164 +174,141 @@ func StartXDSServer(stateDir string) *XDSServer {
+									Cluster: clusterName,
+								},
+								Timeout: &duration.Duration{Seconds: requestTimeout},
+								//IdleTimeout: &duration.Duration{Seconds: idleTimeout},
```
It might be worth noting here that this is set below but I was able to notice it pretty quickly so it's probably fine anyway.

jrajahalme committed:
envoy: Use TypedConfig for Envoy filters

Generic protobuf structpb config encoding is deprecated by
Envoy. Change to using TypedConfig to avoid flooding the logs with
deprecation warnings.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
